server/app/services/program/ProgramServiceImpl.java

```diff
@@ -788,6 +788,11 @@ public ProgramDefinition setProgramQuestionDefinitionAddressCorrectionEnabled(
           "Unexpected error: updating a non address question with address correction enabled");
     }
 
+    if (blockDefinition.hasAddressCorrectionEnabledOnDifferentQuestion(questionDefinitionId)) {
+      throw new ProgramQuestionDefinitionInvalidException(
```

Comment:
@bion I noticed there was a recent changed that uses `BadRequestException` for something like this.  Do you think I should just use that here rather than create a new exception?

Reply:
If there's a chance that the exceptional situation should be handled some other way than returning a 400 then go ahead and make a specific checked exception type for it, otherwise feel free to throw the `BadRequestException`.

A good thing to know about with error handling is this thing: https://github.com/civiform/civiform/blob/main/server/app/controllers/ErrorHandler.java

Reply:
I forgot I'm catching this specifically in AdminProgramBlockQuestionsController so maybe on second thought, I'll leave it.